packages/metrics/src/opentel/global.rs

```diff
@@ -0,0 +1,120 @@
+use core::fmt;
```

Comment:
I understand why you did this, it was to pass only a reference of `meter_provider (MeterProvider)` so that you could use the `meter_provider` later to flush the last remaining metrics before exiting the app.

Although I understand the problem, I would highly recommend against keeping a custom version of the `opentelemetry::global`. Even more because we are creating a library and not an app.

This brings us the unneccessary dependency when something changes with future versions of this `global` in the official library, we have to sync it here. 

There was never needed actually, because if you look at the `MeterProvider` 

![image](https://github.com/kindredgroup/talos/assets/118979108/3e61e7c4-8ceb-4ada-972d-1b6dbbc2a323)

The properties in `MeterProvider` are `Arc`. So all you had to do is just clone the `meter_provider` in your code, before passing to the global. Because the properties in both the clones point to the same inner items. 

![image](https://github.com/kindredgroup/talos/assets/118979108/c39902be-0872-4b5e-9aff-4ad16c499242)

![image](https://github.com/kindredgroup/talos/assets/118979108/c3561d30-d469-4bac-b4ca-d26ecf090e9d)

Here is a snippet to prove they point to the same thing
https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=ba9c6dd2e79a52758d3ed9826bc6fbb6

Reply:
> This brings us the unneccessary dependency when something changes with future versions of this global in the official library, we have to sync it here.

Can you please explain what dependency on global do we have? As far as I can see this is only related to original opentel through MeterProvider trait via official open tel api (look at the imports section at the top of the file).  If they change their global, then it will not affect us. 

Treat it as our version of global :) The useful part here is that we can store anything. MeterProvider is one example, the scaling factors is another. Happy to remove MeterProvider from here and apply cloning, your example suggests that it will work. But then we will end up with - for MP we use original global, for scaling - this one. Is it not better to use one for both purposes?

Reply:
Sorry I have to disagree again with regards to the scaling factor. I assumed the `static GLOBAL_SCALING_FACTOR` variable was available in the `opentelemetry_api::global` package as well and didn't realise that you had introduced this new variable.

I would discourage introducing a global variable just because that is more convenient. I haven't looked too much it depth, but I feel there must be better alternatives. 

- Why not set the scaling factor in a Hashmap, and use an `Arc` and pass it along to where it is required? 
     - I don't see the value of having it as a global variable rather than the convenience to access it.
  
Actually as I look deeper into the code, I would probably not recommend that as well. We know the metrics we are capturing in `cohort-sdk`. So why not let `cohort-sdk` accept the scaling factor as an input `hashmap` when it is created? Probably this would allow us to avoid using `global` altogether in the sdk? 

Why i disagree on the usage of `global`  in the current approach is because `cohort-sdk` which is a library is sharing a global variable with the binary `cohort_banking_with_sdk`, which I think isn't right